### PR TITLE
Add "all" tensor serde schema

### PR DESCRIPTION
**Pull request by vvmnnnkv**

### CODE REVIEW

File: syft/serde/torch_serde.py [hunk 1]
```diff
@@ -31,23 +52,39 @@ def _serialize_tensor(worker: AbstractWorker, tensor) -> bin:
...
+        return numpy_tensor_serializer(tensor)
+    elif worker.serializer == "all":
+        return generic_tensor_serializer(worker, tensor)
     else:
-        raise NotImplementedError("Can't serialize torch tensor without the torch.pickler")
-        # TODO: add another serializer if worker.serializer == 'all' --> @Vova
-        # TODO: Also good place to call numpy_tensor_serializer if worker.serializer == 'numpy'
+        raise NotImplementedError(
+            f"Tensor serialization schema is not supported: {worker.serizalizer}"
+        )
```
Comment (LaRiffle) [inline syft/serde/torch_serde.py]: You can use also a Dict like in `_deserialize_tensor`

File: syft/serde/torch_serde.py [hunk 2]
```diff
@@ -140,7 +200,15 @@ def _simplify_torch_tensor(worker: AbstractWorker, tensor: torch.Tensor) -> bin:
...
         tags = list(tags)
-    return (tensor.id, tensor_bin, chain, grad_chain, tags, tensor.description)
+    return (
+        tensor.id,
+        tensor_bin,
+        chain,
+        grad_chain,
+        tags,
+        tensor.description,
+        syft.serde._simplify(worker, worker.serializer),
```
Comment (LaRiffle) [inline syft/serde/torch_serde.py]: Don't need to simplify it's suppoed to be a string

File: syft/workers/base.py [hunk 3]
```diff
@@ -1005,7 +1005,8 @@ def serializer(self, workers=None):
             workers.append(self)
         else:
             # self is referenced in self.__known_workers
-            workers = [w for _, w in self._known_workers.items()]
+            # NOTE: for some reason self._known_workers may contain a Plan
```
Comment (LaRiffle) [inline syft/workers/base.py]: Well caught! Indeed this happens. Maybe BaseWorker -> AbstractWorker to be more generic?

File: syft/serde/torch_serde.py [hunk 4]
```diff
@@ -31,23 +52,39 @@ def _serialize_tensor(worker: AbstractWorker, tensor) -> bin:
...
+    elif worker.serializer == "numpy":
+        return numpy_tensor_serializer(tensor)
+    elif worker.serializer == "all":
+        return generic_tensor_serializer(worker, tensor)
     else:
-        raise NotImplementedError("Can't serialize torch tensor without the torch.pickler")
-        # TODO: add another serializer if worker.serializer == 'all' --> @Vova
-        # TODO: Also good place to call numpy_tensor_serializer if worker.serializer == 'numpy'
+        raise NotImplementedError(
+            f"Tensor serialization schema is not supported: {worker.serizalizer}"
```
Comment (LaRiffle) [inline syft/serde/torch_serde.py]: typo: "serizalizer" -> serializer

File: syft/serde/torch_serde.py (hunk 2 above)
Comment (vvmnnnkv) [reply syft/serde/torch_serde.py]: Well strings are also simplified to `(5, (b'blah',))`

Comment (LaRiffle) [reply syft/serde/torch_serde.py]: Agreed

File: syft/codes.py [hunk 5]
```diff
@@ -16,6 +16,13 @@ class PLAN_CMDS(object):
     FETCH_PROTOCOL = "fetch_protocol"
 
 
+class TENSOR_SERIALIZATION(object):
+    TORCH = "torch"
```
Comment (LaRiffle) [inline syft/codes.py]: In the future we need to turn these to be integer to have faster serde, but maybe for debug it's convenient to keep it like this

**Pull request merged**